nucliadb/src/nucliadb/common/datamanagers/resources.py

```diff
@@ -101,8 +101,10 @@ async def modify_slug(txn: Transaction, *, kbid: str, rid: str, new_slug: str) -
 
 
 @backoff.on_exception(backoff.expo, (Exception,), jitter=backoff.random_jitter, max_tries=3)
-async def get_resource_shard_id(txn: Transaction, *, kbid: str, rid: str) -> Optional[str]:
-    shard = await txn.get(KB_RESOURCE_SHARD.format(kbid=kbid, uuid=rid))
+async def get_resource_shard_id(
+    txn: Transaction, *, kbid: str, rid: str, for_update: bool = False
```

Comment:
should default be to True, to be safer?

Reply:
although maybe it's ok as we're only really changing this key on rebalance and on resource creation, right?

Reply:
Yeah, I felt comfortable defaulting to false since this is only ever updated in rebalance, which I think it's obvious enough. In resource creation the key is created, so this is never called.